core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
@jedws Ah there was quite a bit of talk on the location of that particular function during this past weekend at Scala by the Bay :-)

I'd like to avoid having such a discussion on this particular ticket, but if it's OK with you perhaps we can chat somewhere else about this? I'd be very interested in hearing your thoughts about it. I could ping you on IRC, or if you're on Gitter that works too.